platforms/nuttx/src/px4/microchip/mpfs/version/board_mcu_version.c

```diff
@@ -34,18 +33,41 @@
 
 /**
  * @file usr_mcu_version.c
- * Implementation of generic user-space version API
+ * Implementation of MPFS version API
  */
 
 #include <px4_platform_common/px4_config.h>
 #include <px4_platform_common/defines.h>
+#include <stdint.h>
 
 #include "board_config.h"
 
-static int hw_version = HW_INFO_INIT_VER;
-static int hw_revision = HW_INFO_INIT_REV;
+#define MPFS_SYS_SERVICE_CR        0x37020050
+#define MPFS_SYS_SERVICE_SR        0x37020054
+#define MPFS_SYS_SERVICE_MAILBOX   0x37020800
+#define SERVICE_CR_REQ             (1 << 0)
+#define SERVICE_SR_BUSY            (1 << 1)
+
+#define getreg8(a)                 (*(volatile uint8_t *)(a))
+#define getreg32(a)                (*(volatile uint32_t *)(a))
+#define putreg32(v,a)              (*(volatile uint32_t *)(a) = (v))
+
+#define MPFS_ICICLE                0x0
```

Comment:
make format

Reply:
I'd rather not have these in here, since they are build-target specific, and this code should fit to any board which is built around polarfire soc